From 9d74a6e3119f9f75638437ab83d35dbaa0c4c5c1 Mon Sep 17 00:00:00 2001
From: Radha Mohan Chintakuntla <radhac@marvell.com>
Date: Tue, 25 May 2021 21:30:11 -0700
Subject: [PATCH 1558/1921] octeontx2-af: Fix ending channel number calculation
 for SDP VF

There is a logic to verify if the channel falls in the SDP channel
range. A minor mistake in calculating the ending channel was causing
failure for a case where we use all the SDP channels and last VF tries
to allocate the last channel. This patch fixes the mistake.

Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
Change-Id: Iff251fe2abcf305b7df60183fe541389e436d2bd
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/52671
Reviewed-by: Satananda Burla <sburla@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <sgoutham@marvell.com>
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
index 1a0cd3066b19..d5e865725875 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_nix.c
@@ -337,7 +337,7 @@ static int nix_interface_init(struct rvu *rvu, u16 pcifunc, int type, int nixlf,
 			for (vfid = 0; vfid < vf; vfid++)
 				req_chan_base += sdp_info->vf_rings[vfid];
 			req_chan_cnt = sdp_info->vf_rings[vf];
-			req_chan_end = req_chan_base + req_chan_cnt;
+			req_chan_end = req_chan_base + req_chan_cnt - 1;
 			if (req_chan_base < rvu_nix_chan_sdp(rvu, 0) ||
 			    req_chan_end > rvu_nix_chan_sdp(rvu, 255)) {
 				dev_err(rvu->dev,
-- 
2.31.1

